File: tests/openmp_examples/devices/array_shaping.1.f90
!$omp          declare target enter(do_work)
!$omp          declare target enter(other_work)
!$omp    target data map(a)
!$omp       target
!$omp       end target
!$omp       target update from( a(1,1:ny), a(nx,1:ny) )
!$omp       target update to( a(0,1:ny), a(nx+1,1:ny) )
!$omp       target
!$omp       end target
!$omp    end target data

File: tests/openmp_examples/devices/array_shaping.1.c
#pragma omp begin declare target
#pragma omp end declare target
#pragma omp target data map(a[0:nx*(ny+2)])
#pragma omp target
#pragma omp target update from( (([nx][ny+2])a)[0:nx][1], (([nx][ny+2])a)[0:nx][ny] )
#pragma omp target update to( (([nx][ny+2])a)[0:nx][0], (([nx][ny+2])a)[0:nx][ny+1] )
#pragma omp target
